### PR TITLE
202004 query improvement

### PR DESCRIPTION
**Pull request by alaurenz**
Trello:
https://trello.com/c/Sp88aKwl/207-totago-tpl-data-ingest-osm-data-pipeline-mvp-for-se-michigan

### CODE REVIEW

**Comment by alaurenz** [comment]
[TOTAGO/TPL - Data ingest: OSM Data Pipeline MVP for SE Michigan](https://trello.com/c/Sp88aKwl/207-totago-tpl-data-ingest-osm-data-pipeline-mvp-for-se-michigan)

File: geoPipe.py [hunk 1]
```diff
@@ -26,9 +30,121 @@
...
+		return '[out:json][timeout:25]; \
+			(\
+				way["highway"~"path|footway|footpath|bridleway"]\
+				["footway"!~"sidewalk|crossing"]\
+				["bicycle"~"yes|designated"]\
+				(poly:"{}");\
+				way["highway"~"cycleway"]\
+				(poly:"{}");\
+			);(._;>;);out;'.format(polygon, polygon)
+
```
Comment (axme100) [inline geoPipe.py:100]: Hmm so you're just manually formatting the JSON being returned here?: I haven't seen syntax like this.

File: geoPipe.py [hunk 2]
```diff
@@ -341,41 +458,27 @@ def to_db(trail_df, region_code, tablename, schema=""):
...
-			3) For each node in each way, replace the node ID with an dict object of: ID, lat, lon
-			4) Names each trail from tags
-	"""
-	print("Requesting trails for {}".format(REGION))
+	print("\nSleeping for " + str(SLEEP_BETWEEN_QUERIES_SEC) + " secs")
+	time.sleep(SLEEP_BETWEEN_QUERIES_SEC)
 
+	print("\nquerying ways from OSM")
 
-	# tqdm means "progress" in Arabic, this guy wraps iterables and predicts the time it'll take to run.
```
Comment (axme100) [inline geoPipe.py:368]: Wait did I write this, hahaha? Did you know it is also stands for "Te quiero demasiado" in Spanish

File: geoPipe.py [hunk 3]
```diff
@@ -436,21 +535,110 @@ def main():
...
 
-	print("found " + str(len(trail_df)) + " trails in " + REGION)
+	return len(trail_df)
+	
+def get_region_code(park_id, park_index):
+	return TYPE + "_" + park_id + "_" + str(park_index)
+
+def main():
+
+	sys.setrecursionlimit(10000)
```
Comment (axme100) [inline geoPipe.py:558]: Wait where are you using recursion in this code?

File: geoPipe.py (hunk 1 above)
Comment (alaurenz) [reply geoPipe.py:102]: No, this is actually a query to OpenStreetMap database for trails. Then the response is set to come back as JSON. The .format is just a way to put variables into a Python string, nothing fancy, just the same as "str1 " + variable + " str2"

File: geoPipe.py (hunk 2 above)
Comment (alaurenz) [reply geoPipe.py:368]: No haha, this code was written by one of our interns from the University of Michigan.  They actually wrote this pipeline originally, I just updated the code a bit.

File: geoPipe.py (hunk 3 above)
Comment (alaurenz) [reply geoPipe.py:560]: ways_to_trails method is recursive, but this was code from before, I did not add any new recursion in this PR, just moved methods around.

**Pull request merged**